File: frontend/src/types/task.ts
export type TaskStatus = 'backlog' | 'todo' | 'in_progress' | 'review' | 'done' | 'archived';
export type TaskPriority = 'none' | 'low' | 'medium' | 'high' | 'urgent';
export type WorkspaceType = 'personal' | 'enterprise';
export type RecurrenceType = 'daily' | 'weekly' | 'monthly' | 'yearly';

export interface Recurrence {
  id: string;
  type: RecurrenceType;
  interval: number;
  days_of_week: number[] | null;
  end_date: string | null;
  max_occurrences: number | null;
}

export interface Task {
  id: string;
  title: string;
  description: string | null;
  notes: string | null;
  status: TaskStatus;
  priority: TaskPriority;
  workspace: WorkspaceType;
  enterprise_id: string | null;
  creator_id: string;
  responsible_id: string | null;
  is_public: boolean;
  planned_start_at: string | null;
  started_at: string | null;
  due_at: string | null;
  completed_at: string | null;
  created_at: string;
  updated_at: string;
  assignee_ids: string[];
  recurrence: Recurrence | null;
}

export interface TaskCreate {
  title: string;
  description?: string | null;
  notes?: string | null;
  status?: TaskStatus;
  priority?: TaskPriority;
  workspace?: WorkspaceType;
  enterprise_id?: string | null;
  responsible_id?: string | null;
  is_public?: boolean;
  planned_start_at?: string | null;
  due_at?: string | null;
  recurrence?: {
    type: RecurrenceType;
    interval?: number;
    days_of_week?: number[] | null;
    end_date?: string | null;
    max_occurrences?: number | null;
  } | null;
}

export interface TaskUpdate {
  title?: string;
  description?: string | null;
  notes?: string | null;
  status?: TaskStatus;
  priority?: TaskPriority;
  responsible_id?: string | null;
  is_public?: boolean;
  planned_start_at?: string | null;
  due_at?: string | null;
}

export interface TaskFilters {
  status?: TaskStatus;
  priority?: TaskPriority;
  search?: string;
}

export const TASK_STATUS_LABELS: Record<TaskStatus, string> = {
  backlog: 'Backlog',
  todo: 'A Fazer',
  in_progress: 'Em Andamento',
  review: 'Em Revisão',
  done: 'Concluída',
  archived: 'Arquivada',
};

export const TASK_PRIORITY_LABELS: Record<TaskPriority, string> = {
  none: 'Nenhuma',
  low: 'Baixa',
  medium: 'Média',
  high: 'Alta',
  urgent: 'Urgente',
};

export const TASK_STATUS_ORDER: TaskStatus[] = [
  'backlog', 'todo', 'in_progress', 'review', 'done', 'archived',
];

File: backend/app/schemas/task.py
from typing import Optional, List
from datetime import datetime
from pydantic import BaseModel

from app.models.enums import (
    TaskStatus, TaskPriority, WorkspaceType, RecurrenceType
)


# ── Recurrence ────────────────────────────────────────────────────────────────

class RecurrenceCreate(BaseModel):
    type: RecurrenceType
    interval: int = 1
    days_of_week: Optional[List[int]] = None
    end_date: Optional[datetime] = None
    max_occurrences: Optional[int] = None


class RecurrenceRead(BaseModel):
    id: str
    type: RecurrenceType
    interval: int
    days_of_week: Optional[List[int]]
    end_date: Optional[datetime]
    max_occurrences: Optional[int]

    class Config:
        from_attributes = True


# ── Task ─────────────────────────────────────────────────────────────────────

class TaskCreate(BaseModel):
    title: str
    description: Optional[str] = None
    notes: Optional[str] = None
    status: TaskStatus = TaskStatus.todo
    priority: TaskPriority = TaskPriority.none
    workspace: WorkspaceType = WorkspaceType.personal
    enterprise_id: Optional[str] = None
    responsible_id: Optional[str] = None
    is_public: bool = False
    planned_start_at: Optional[datetime] = None
    due_at: Optional[datetime] = None
    recurrence: Optional[RecurrenceCreate] = None


class TaskUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    notes: Optional[str] = None
    status: Optional[TaskStatus] = None
    priority: Optional[TaskPriority] = None
    responsible_id: Optional[str] = None
    is_public: Optional[bool] = None
    planned_start_at: Optional[datetime] = None
    due_at: Optional[datetime] = None


class TaskStatusUpdate(BaseModel):
    status: TaskStatus


class TaskRead(BaseModel):
    id: str
    title: str
    description: Optional[str]
    notes: Optional[str]
    status: TaskStatus
    priority: TaskPriority
    workspace: WorkspaceType
    enterprise_id: Optional[str]
    creator_id: str
    responsible_id: Optional[str]
    is_public: bool
    planned_start_at: Optional[datetime]
    started_at: Optional[datetime]
    due_at: Optional[datetime]
    completed_at: Optional[datetime]
    created_at: datetime
    updated_at: datetime
    assignee_ids: List[str] = []
    recurrence: Optional[RecurrenceRead] = None

    class Config:
        from_attributes = True


class TaskFilters(BaseModel):
    status: Optional[TaskStatus] = None
    priority: Optional[TaskPriority] = None
    search: Optional[str] = None
    enterprise_id: Optional[str] = None
    workspace: Optional[WorkspaceType] = None
